sdk/provisioning/Azure.Provisioning/src/Expressions/BicepTypeMapping.cs

```diff
@@ -96,6 +98,11 @@ public static BicepExpression ToBicep(object? value, string? format)
             // in bicep source and you need to use a parameter file for larger
             // values
             long i => BicepSyntax.Value((int)i),
+            // Note: bicep does not offically support floating numbers
+            // therefore for floating numbers we are taking a workaround from
+            // https://github.com/Azure/bicep/issues/1386#issuecomment-818077233
+            float f => FromFloat(f),
+            double d => FromDouble(d),
```

Comment:
I think we can get away with sharing `FromDouble` so we have less code:
```suggestion
            float f => FromDouble(f),
            double d => FromDouble(d),
```